sql/insert.go

```diff
@@ -186,3 +224,94 @@ func (p *planner) processColumns(tableDesc *TableDescriptor,
 
 	return cols, nil
 }
+
+func (p *planner) makeDefaultExprs(cols []ColumnDescriptor) ([]parser.Expr, error) {
+	// Check to see if any of the columns have DEFAULT expressions. If there are
+	// no DEFAULT expressions, we don't bother with constructing the defaults map
+	// as the defaults are all NULL.
+	haveDefaults := false
+	for _, col := range cols {
+		if col.DefaultExpr != nil {
+			haveDefaults = true
+			break
+		}
+	}
+	if !haveDefaults {
+		return nil, nil
+	}
+
+	// Construct a SELECT statement with an output for each column. The output is
+	// either the DEFAULT expression or NULL if no default expression was
+	// configured.
+	var buf bytes.Buffer
+	buf.WriteString("SELECT ")
+	for i, col := range cols {
+		if i > 0 {
+			buf.WriteString(", ")
+		}
+		if col.DefaultExpr != nil {
+			buf.WriteString(*col.DefaultExpr)
+		} else {
+			buf.WriteString("NULL")
+		}
+	}
+
+	sql := buf.String()
+	stmts, err := parser.Parse(sql, parser.Syntax(p.session.Syntax))
```

Comment:
I was thinking about building up the select statement, and by building the AST directly I meant to parse each expression individually and then build a select statement from that, without ever appending commas to a buffer. (of course that wouldn't actually make sense here because you don't care about the select statement; you're just using it as a way to get to the expression parser). Could we have a `parser.ParseExpr` function to use here?